credsweeper/deep_scanner/html_scanner.py

```diff
@@ -19,7 +19,8 @@ def data_scan(
             depth: int,  #
             recursive_limit_size: int) -> List[Candidate]:
         """Tries to represent data as html text and scan as text lines"""
-        if data_provider.represent_as_html(depth, recursive_limit_size):
+        if data_provider.represent_as_html(depth, recursive_limit_size,
+                                           self.scanner.keywords_required_substrings_check):
```

Comment:
Other changes are looking good.
But I couldn't understand why `represent_as_html()` should to filter the keywords with required sub-strings... :thinking: 
> `HTML representation : collect matched keywords fields only`

Reply:
The representation combine header of table with each cell in the column (--depth > 2 option).
To reduce extra combinations i put the check. If  a header has no any substring - there is no sense do the combination